### PR TITLE
chore(profiling): use mock agent for profiling tests

### PR DESCRIPTION
**Pull request by gyuheon0h**
note: I recognize there are a gazillion failing tests because of tiny edge cases and also how different python versions treat these type annotations... fixing soon...

## Description
Previously, profiling tests captured profiles by writing .pprof files to disk using `DD_PROFILING_OUTPUT_PPROF`, then reading them back with `parse_newest_profile()`. 

This PR introduces a self-contained `_ProfilingMiniAgent` HTTP server that receives profiles over the same multipart upload path that production uses. Each test gets a unique UUID session token; `ddup` is configured to POST to `{base_url}/session/{token}/profiling/v1/input`.

An autouse profiling_test_agent fixture in `tests/profiling/conftest.py` starts the server and sets `_DD_PROFILING_TEST_TOKEN` / `_DD_PROFILING_TEST_PROXY_URL` in the environment before each test, so subprocess tests inherit them automatically without any per-test boilerplate.

Tests retrieve profiles by `pprof_utils.get_profile_from_agent()`, which polls the mini agent until an upload appears and parses the multipart response.
 
`config_test_token()` is added to `_ddup.pyx` to update the routing at runtime for in-process tests (where ddup is already imported before the fixture sets env vars), and `_get_endpoint()` checks the token before falling back to the real agent URL so production behavior is unaffected.

### CODE REVIEW

**Comment by gyuheon0h** [comment]
* **#18877** <a href="https://app.graphite.com/github/pr/DataDog/dd-trace-py/18877?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/DataDog/dd-trace-py/18877?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by taegyunkim — COMMENTED** [review]
Do you mind addressing the codex review comments? 

Also, let's make sure to have a test using `DD_PROFILING_OUTPUT_PPROF` as we might rely on that for some of our local/one-off testing use cases

File: ddtrace/internal/datadog/profiling/dd_wrapper/src/uploader_builder.cpp [hunk 1]
```diff
@@ -101,9 +101,7 @@ Datadog::UploaderBuilder::set_process_tags(std::string_view p_tags)
 void
 Datadog::UploaderBuilder::set_output_filename(std::string_view _output_filename)
 {
-    if (!_output_filename.empty()) {
-        ProfilerState::get().output_filename = _output_filename;
-    }
+    ProfilerState::get().output_filename = _output_filename;
```
Comment (taegyunkim) [inline ddtrace/internal/datadog/profiling/dd_wrapper/src/uploader_builder.cpp]: What was the reasoning behind dropping this if check?

File: ddtrace/internal/datadog/profiling/ddup/_ddup.pyx [hunk 2]
```diff
@@ -371,8 +380,13 @@ def config(
         call_func_with_str(ddup_config_env, env)
     if version:
         call_func_with_str(ddup_config_version, version)
+    # Always call ddup_config_output_filename even when output_filename is empty so that
+    # a previous file-output configuration is cleared.  call_func_with_str skips falsy
+    # strings, so call the C function directly with an empty string_view here.
```
Comment (taegyunkim) [inline ddtrace/internal/datadog/profiling/ddup/_ddup.pyx]: why?

Comment (gyuheon0h) [reply ddtrace/internal/datadog/profiling/ddup/_ddup.pyx]: This (also your other comment) were left overs from my first iteration where I did not cleanly configure where the output goes. Reverted both and did another pass to catch similar stale changes that I missed.

File: .github/workflows/pytorch_gpu_tests.yml
Comment (KowalskiThomas) [inline .github/workflows/pytorch_gpu_tests.yml:1]: Is this PR ready for review? Given that it's mainly changing the testing infrastructure and tests are failing, would you mind re-asking for review once CI actually passes?
